### PR TITLE
js_promise_function_DOM: Cypress tests added

### PR DESCRIPTION
**Pull request by iBrianWarner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 Create function `waitFor` which takes two parameters:
-- `element` - DOM element
-- `eventName`
+- `element` - DOM element;
+- `eventName`, such as `click`, `input` and `blur`.
 
 The function returns a Promise which should be resolved when `eventName` happened on `element`.
 
-Hint: It doesn't matter in what way you choose to notify users about resolved or rejected promises. You can use DOM or `console.log`.
+Example of the promise: `It was click on the element: INPUT, id: login.`
```
Comment (roman-mate) [inline README.md:17]: Not sure I understand this. Please rephrase or we can talk about this during the standup.

File: README.md [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
+
+Hint: You have to use DOM to notify users about resolved or rejected promises.
```
Comment (roman-mate) [inline README.md:19]: How should I use DOM to notify users?

File: cypress/integration/promiseFunction.spec.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+Cypress.Commands.add('hasPromise', (promise, element, id) => {
```
Comment (roman-mate) [inline cypress/integration/promiseFunction.spec.js:3]: Pormise is a javascript object with three states: pending, resolved, rejected. 

How can page 'havePromise'? Let's chat offline about a better name.

Comment (iBrianWarner) [reply cypress/integration/promiseFunction.spec.js:3]: renamed

**Review by roman-mate — COMMENTED** [review]
did you forget to push the changes? There are unresolved comments.

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+const promiseBlur3 = waitFor(button, 'blur');
+
+promiseClick1.then(printMessage);
+promiseClick2.then(printMessage);
+promiseClick3.then(printMessage);
+promiseInput1.then(printMessage);
+promiseInput2.then(printMessage);
+promiseBlur1.then(printMessage);
+promiseBlur2.then(printMessage);
+promiseBlur3.then(printMessage);
```
Comment (iBrianWarner) [inline src/scripts/main.js:30]: I don't like how it looks, but don't know how to improve

File: README.md (hunk 1 above)
Comment (iBrianWarner) [reply README.md]: changed description

File: README.md [hunk 5]
```diff
@@ -1,16 +1,20 @@
...
 
 To complete this task, you should create a function that returns a `Promise`.
 Create function `waitFor` which takes two parameters:
-- `element` - DOM element
-- `eventName`
+- `element` - DOM element;
+- `eventName`, such as `click`, `input` and `blur`.
 
 The function returns a Promise which should be resolved when `eventName` happened on `element`.
+Function should print the message with text: `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`
```
Comment (roman-mate) [inline README.md:16]: What does it mean to `print the message`? console.log?

File: cypress/integration/promiseFunction.spec.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+Cypress.Commands.add('resolvePromise', (promise, element, id) => {
```
Comment (roman-mate) [inline cypress/integration/promiseFunction.spec.js:3]: need a better name.

File: src/index.html [hunk 7]
```diff
@@ -27,7 +27,7 @@
          >
       </label>
 
-      <button type="submit" id="submit">Login</button>
+      <button type="button" id="submit">Login</button>
```
Comment (roman-mate) [inline src/index.html:30]: why make this change?

File: src/scripts/main.js (hunk 4 above)
Comment (roman-mate) [reply src/scripts/main.js:30]: create a function initPromises. move logic into the function. call it in the end. 

login => loginElement

waitFor(login, 'click').then(printMessage);
....

File: cypress/integration/promiseFunction.spec.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+Cypress.Commands.add('checkNotificationExist', (promise, element, id) => {
```
Comment (roman-mate) [inline cypress/integration/promiseFunction.spec.js:3]:
```suggestion
Cypress.Commands.add('checkNotificationExists', (promise, element, id) => {
```

Comment (roman-mate) [reply cypress/integration/promiseFunction.spec.js:3]: promise is an object. it looks like we are referring event.

**Review by roman-mate — COMMENTED** [review]
final nits.

File: README.md [hunk 9]
```diff
@@ -1,16 +1,24 @@
...
-- `element` - DOM element
-- `eventName`
+- `element` - DOM element;
+- `eventName`, such as `click`, `input` and `blur`.
+The function returns a `Promise` which should be resolved when `eventName` happened on `element`.
+Function should return the notification message with text:
+`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`
+For example: `It was click on the element: INPUT, id: login.`
+
+Create a function `printMessage` which takes one parameter `message` and returns a <div> element with notification message above. Notification message should have class `message`.
```
Comment (roman-mate) [inline README.md:20]:
```suggestion
Create a function `printMessage` which takes one parameter `message` and adds to the DOM a <div> element with notification message above. Notification message should have class `message`.
```

File: cypress/integration/promiseFunction.spec.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  });
+
+  it(`should resolve 'input' promise for 'login' field`, () => {
+    cy.get('#login').type('login');
+    cy.checkNotificationExists('input', 'INPUT', 'login');
+  });
+
+  it(`should resolve 'blur' promise for 'login' field`, () => {
+    cy.get('#login').click();
+    cy.get('body').click({ force: true });
```
Comment (roman-mate) [inline cypress/integration/promiseFunction.spec.js:25]: why not https://docs.cypress.io/api/commands/blur.html#Usage ?

**Review by roman-mate — APPROVED** [review]
Final nits.

File: cypress/integration/promiseFunction.spec.js (hunk 10 above)
Comment (iBrianWarner) [reply cypress/integration/promiseFunction.spec.js]: done!

**Pull request merged**